File: src/android/app/src/main/java/org/citra/citra_emu/utils/Log.java
package org.citra.citra_emu.utils;

import org.citra.citra_emu.BuildConfig;

/**
 * Contains methods that call through to {@link android.util.Log}, but
 * with the same TAG automatically provided. Also no-ops VERBOSE and DEBUG log
 * levels in release builds.
 */
public final class Log {
    private static final String TAG = "Citra Frontend";

    private Log() {
    }

    public static void verbose(String message) {
        if (BuildConfig.DEBUG) {
            android.util.Log.v(TAG, message);
        }
    }

    public static void debug(String message) {
        if (BuildConfig.DEBUG) {
            android.util.Log.d(TAG, message);
        }
    }

    public static void info(String message) {
        android.util.Log.i(TAG, message);
    }

    public static void warning(String message) {
        android.util.Log.w(TAG, message);
    }

    public static void error(String message) {
        android.util.Log.e(TAG, message);
    }
}

File: src/android/app/src/main/java/org/citra/citra_emu/utils/Log.kt
// Copyright 2023 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

package org.citra.citra_emu.utils

object Log {
    // Tracks whether we should share the old log or the current log
    var gameLaunched = false

    external fun debug(message: String)

    external fun warning(message: String)

    external fun info(message: String)

    external fun error(message: String)

    external fun critical(message: String)
}
